src/api/feed-discovery/test/router.test.js

```diff
@@ -0,0 +1,153 @@
+const request = require('supertest');
+const nock = require('nock');
+const createError = require('http-errors');
+const { app } = require('../src');
+const { checkValidUrl, checkValidBlogPage, discoverFeedUrls } = require('../src/lib');
+
+describe('POST /', () => {
+  afterAll(() => {
+    nock.cleanAll();
+  });
+  describe('Test checkValidUrl middleware', () => {
+    it('should return 400 status if the url has invalid format', (done) => {
+      const invalidUrl = 'invalidLink.com';
+      const mockReq = {
+        body: {
+          blogUrl: invalidUrl,
+        },
+      };
+      const mockRes = {};
+      const mockNextFunction = jest.fn();
+      checkValidUrl(mockReq, mockRes, mockNextFunction);
+      const expectedError = createError(400, 'Invalid Blog URL');
+
+      // Expect the next() function to be called with the expected error
+      expect(mockNextFunction).toBeCalledWith(expectedError);
+
+      request(app)
+        .post('/')
+        .send({ blogUrl: invalidUrl })
+        // Expect to receive 400 status and expected response body
+        .expect(400, '<h1>400 Error</h1><p>Invalid Blog URL</p>', done);
```

Comment:
Let's not depend on the actual HTML that gets returned, just the 400.  We might change it later.